### PR TITLE
[wip] Add the Wifi Problem Reporting screen

### LINKED ISSUE #934: Add a "Report a WiFi Problem" quick-access button
> - https://github.com/mastermoo/react-native-action-button
> - https://github.com/maxs15/react-native-modalbox

I'm envisioning a FAB on the homescreen, in the bottom-right, that gives you two options when you tap it:

- Report a WiFi Problem
- [and maybe] Make a Helpdesk Ticket

This issue is about the framework of the button and the Report WiFi button, though.

---

I think that when you tap the Wifi option, it should pop open the modal and say "Collecting data…".

We need to submit the user's device info, as well as the current location (lat/long/alt).

I am making an endpoint to store the data: we'll be able to POST a json blob to www.stolaf.edu/apps/all-about-olaf/wifi when I'm done there (nothing exists yet.)

### PR DESCRIPTION
**Pull request by hawkrives**
> this will close #934 

This currently collects _most_ of the data needed in order to submit to the wifi problem reporting backend.

screenshot of data reporting screen

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
-    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/> <!--  Precise location to use promptLocation for letting OneSignal know the user location. -->
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/> <!-- Precise location to use promptLocation for letting OneSignal know the user location. -->
```
Comment (drewvolz) [inline android/app/src/main/AndroidManifest.xml]: Comment might want to be tweaked towards this being used for wifi reporting.

File: source/views/help/wifi.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    return (
+      <View>
+        <Card header="Report a WiFi Problem!" footer={this.state.status}>
+          <Button
+            disabled={this.state.status !== ''}
+            onPress={this.start}
+            title="Report"
+          />
+          <Text>
+            {this.state.data && JSON.stringify(this.state.data, null, 2)}
```
Comment (drewvolz) [inline source/views/help/wifi.js:96]: What's going on with the arguments ... `null` and `2`?

**Comment by drewvolz** [comment]
Overall -- this is a very nice set of changes. Well done! I am looking forward to seeing what this becomes.

**Comment by drewvolz** [comment]
Your screenshot shows an iOS device. Is this functionality x-platform? I thought we were only able to gather specific network information from Android devices.

File: source/views/help/wifi.js (hunk 2 above)
Comment (hawkrives) [reply source/views/help/wifi.js:96]: 2 is how many spaces to indent.

**Comment by hawkrives** [comment]
I'm not gathering network information at all – just device and location.

**Comment by drewvolz** [comment]
Ah, I see. I lumped a few attributes of the location into network information in my head. Thanks.

**Comment by drewvolz** [comment]
Testing this out, disallowing location information will need to prompt the user for instructions on how to enable location services if they are denied the first time.

**Comment by hawkrives** [comment]
Valid.

**Comment by hawkrives** [comment]
This is currently waiting on me to 

1. build the receiving end and 
2. decide how to do request queuing in the app (or drop that feature.)

**Comment by hawkrives** [comment]
I'm going to merge this, but with the screen disabled. That way, we can get the native library into the next release even if I don't finish the screen in time.

**Pull request merged**